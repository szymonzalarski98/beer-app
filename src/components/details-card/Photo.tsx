import React from 'react';
import { View, Image, StyleSheet } from 'react-native';
import { deviceWidth } from '../../styles/helpers.style';

interface Props {
  image_url: string;
};

const Photo: React.SFC<Props> = ({ image_url }) => {
  return (
    <View style={styles.imageContainer}>
      {
        image_url !== null
        ?
        <Image source={{ uri: image_url }} style={styles.image} resizeMode="contain" />
        :
        <View style={[styles.image, { backgroundColor: '#fafafa' }]} />
      }
    </View>
  );
};

export default Photo;

const styles = StyleSheet.create({
  imageContainer: {
    flex: 0.8,
    alignItems: 'center',
    justifyContent: 'center',
  },
  image: {
    flex: 1,
    margin: deviceWidth * 0.04,
    aspectRatio: 1,
  },
});