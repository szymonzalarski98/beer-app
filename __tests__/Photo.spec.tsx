import React from 'react';
import { render } from 'react-native-testing-library';
import Photo from '../src/components/details-card/Photo';

describe('Test Photo component', () => {
  it ('Should match snapshot', () => {
    const { baseElement } = render(<Photo image_url="test" />);
    expect(baseElement).toMatchSnapshot();
  });
});