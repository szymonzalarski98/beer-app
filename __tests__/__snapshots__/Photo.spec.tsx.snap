// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Photo component Should match snapshot 1`] = `undefined`;
